docs: document balancing cores and max-jobs

grahamc opened the pull request:
Write a bit about the balance between max-jobs and cores.

edolstra commented on doc/manual/advanced-topics/cores-vs-jobs.xml:
```diff
@@ -0,0 +1,120 @@
+    <listitem><para>
+      Dictates how many separate derivations will be built at the same
+      time. If you set this to zero, the local machine will attempt to
+      do no builds (forcing it to be built remotely.)
```
`will attempt to do no build` -> `will do no builds`.

edolstra commented on doc/manual/advanced-topics/cores-vs-jobs.xml:
```diff
@@ -0,0 +1,120 @@
+    <listitem><para>
+      Dictates how many cores each derivation is allowed to use. If
+      you set this to zero, the builds will be given access to all the
+      cores of your system.
```
"Dictates" -> "suggests" (and drop the second sentence). There is nothing that stops a derivation from using more cores. In practice, the "cores" option translates to GNU Make's `-j <cores>` flag.

edolstra commented on doc/manual/advanced-topics/cores-vs-jobs.xml:
```diff
@@ -0,0 +1,120 @@
+  </tbody>
+</table>
+
+<para>It is up to the derivations' build scirpt to respect
```
`scirpt` -> `script.

jonringer commented:
ping @grahamc 

I still  think this is really valuable, and would love to have it in the manual :)

grahamc commented:
Thanks @jonringer, I've updated the docs and addressed @edolstra's feedback. Would you might giving it a read for mistakes? I have a tendency to make silly mistakes :x

jonringer commented on doc/manual/advanced-topics/advanced-topics.xml:
```diff
@@ -7,6 +7,8 @@
+<xi:include href="cores-vs-jobs.xml" />
 <xi:include href="diff-hook.xml" />
 
+
```
Extra white space? I don't see 2 newlines anywhere else.
```suggestion
```

grahamc replied:
Thanks! Fixed up!
